subsys/net/lib/websocket/websocket.c

```diff
@@ -419,6 +419,12 @@ static int websocket_interal_disconnect(struct websocket_context *ctx)
 
 	NET_DBG("[%p] Disconnecting", ctx);
 
+	ret = websocket_send_msg(ctx->sock, NULL, 0, WEBSOCKET_OPCODE_CLOSE,
+				 true, true, SYS_FOREVER_MS);
+	if (ret < 0) {
+		NET_ERR("[%p] Failed to send close message (err %d).", ctx, ret);
+	}
+
 	ret = close(ctx->real_sock);
```

Comment:
The return code is overwritten by the close here, is this intended?

Reply:
Yes, IMO `close()` is essential here when it comes to the error code returned, as send could fail in certain cases (for example if the peer has already closed the TCP connection).